Lib/html/parser.py

```diff
@@ -57,6 +58,26 @@
 # </ and the tag name, so maybe this should be fixed
 endtagfind = re.compile(r'</\s*([a-zA-Z][-.a-zA-Z0-9:_]*)\s*>')
 
+# Character reference processing logic specific to attribute values
+# See: https://html.spec.whatwg.org/multipage/parsing.html#named-character-reference-state
+attr_charref = re.compile(r'&(#[0-9]+|#[xX][0-9a-fA-F]+|[a-zA-Z][a-zA-Z0-9]*)[;=]?')
+
+def replace_attr_charref(match):
+    ref = match.group(0)
+    # Numeric / hex char refs must always be unescaped
+    if ref[1] == '#':
+        return unescape(ref)
+    # Named character / entity references must only be unescaped
+    # if they are an exact match, and they are not followed by an equals sign
+    terminates_with_equals = ref[-1:] == '='
+    exact_match = ref.lstrip('&').rstrip('=') in html5_entities
+    if exact_match and not terminates_with_equals:
+        return unescape(ref)
+    # Otherwise do not unescape
+    return ref
+
+def unescape_attrvalue(s):
+    return attr_charref.sub(replace_attr_charref, s)
```

Comment:
Both functions should be private, and their name prefixed by an `_`.

Reply:
Done